commit 909010af44d3cab2066bfe02a916a36fe1040e54
Author: Han <hangenuit@gmail.com>
Date:   Tue Oct 4 14:14:07 2011 +0200

    BUG: Fix version comparison bug on Py25.

diff --git a/numpy/distutils/mingw32ccompiler.py b/numpy/distutils/mingw32ccompiler.py
index 78457bb..b3aa96e 100644
--- a/numpy/distutils/mingw32ccompiler.py
+++ b/numpy/distutils/mingw32ccompiler.py
@@ -115,7 +115,7 @@ class Mingw32CCompiler(distutils.cygwinccompiler.CygwinCCompiler):
         # bad consequences, like using Py_ModuleInit4 instead of
         # Py_ModuleInit4_64, etc... So we add it here
         if get_build_architecture() == 'AMD64':
-            if self.gcc_version < "4.":
+            if self.gcc_version < "4.0":
                 self.set_executables(
                     compiler='gcc -g -DDEBUG -DMS_WIN64 -mno-cygwin -O0 -Wall',
                     compiler_so='gcc -g -DDEBUG -DMS_WIN64 -mno-cygwin -O0 -Wall -Wstrict-prototypes',
@@ -135,7 +135,7 @@ class Mingw32CCompiler(distutils.cygwinccompiler.CygwinCCompiler):
                                      linker_exe='g++ -mno-cygwin',
                                      linker_so='%s -mno-cygwin -mdll -static %s'
                                      % (self.linker, entry_point))
-            elif self.gcc_version < "4.":
+            elif self.gcc_version < "4.0":
                 self.set_executables(compiler='gcc -mno-cygwin -O2 -Wall',
                                      compiler_so='gcc -mno-cygwin -O2 -Wall -Wstrict-prototypes',
                                      linker_exe='g++ -mno-cygwin',
